python/cudf/cudf/core/frame.py

```diff
@@ -983,12 +983,29 @@ def searchsorted(
         1-D cupy array of insertion points
         """
         # Call libcudf++ search_sorted primitive
+
+        from cudf.utils.dtypes import is_scalar
+
+        scalar_flag = None
+        if is_scalar(values):
+            scalar_flag = True
+
+        if not isinstance(values, Frame):
+            values = as_column(
+                values,
+                dtype=self.dtype if isinstance(self, cudf.Series) else None,
```

Comment:
Just curious, do we also need to do this if `isinstance(self, cudf.Index)`?

Reply:
Maybe not because it'd be handled as it is a subclass of `Frame`, related test: https://github.com/rapidsai/cudf/pull/4650/files#diff-e821c981a2faa2a251494e677d34f042R35